src/main/java/com/amazon/opendistroforelasticsearch/sql/optimizer/subquery/rewriter/SubqueryAliasRewriter.java

```diff
@@ -0,0 +1,171 @@
+ * +-----------------------+-------------+-----------------------------------------------------------------------------------------------------+
+ * | (Start)               | ()          | SELECT * FROM TbA WHERE a IN (SELECT b FROM TbB) and c > 10                                         |
+ * +-----------------------+-------------+-----------------------------------------------------------------------------------------------------+
+ * | MySqlSelectQueryBlock | (TbA,TbA_0) | SELECT * FROM TbA as TbA_0 WHERE a IN (SELECT b FROM TbB) and c > 10                                |
+ * +-----------------------+-------------+-----------------------------------------------------------------------------------------------------+
+ * | Identifier in Select  | (TbA,TbA_0) | SELECT TbA.* FROM TbA as TbA_0 WHERE a IN (SELECT b FROM TbB) and c > 10                            |
+ * +-----------------------+-------------+-----------------------------------------------------------------------------------------------------+
+ * | Identifier in Where   | (TbA,TbA_0) | SELECT TbA.* FROM TbA as TbA_0 WHERE TbA_0.a IN (SELECT b FROM TbB) and TbA_0.c > 10                |
+ * +-----------------------+-------------+-----------------------------------------------------------------------------------------------------+
+ * | MySqlSelectQueryBlock | (TbA,TbA_0) | SELECT TbA.* FROM TbA as TbA_0 WHERE TbA_0.a IN (SELECT b FROM TbB as TbB_0) and TbA_0.c > 10       |
+ * |                       | (TbB,TbB_0) |                                                                                                     |
+ * +-----------------------+-------------+-----------------------------------------------------------------------------------------------------+
+ * | Identifier in Select  | (TbA,TbA_0) | SELECT TbA.* FROM TbA as TbA_0 WHERE TbA_0.a IN (SELECT TbB_0.b FROM TbB as TbB_0) and TbA_0.c > 10 |
+ * |                       | (TbB,TbB_0) |                                                                                                     |
+ * +-----------------------+-------------+-----------------------------------------------------------------------------------------------------+
+ *
+ */
+public class SubqueryAliasRewriter extends MySqlASTVisitorAdapter {
+    private final Deque<Table> tableScope = new ArrayDeque<>();
+    private int aliasSuffix = 0;
+    private final static String DOT = ".";
+
+    @Override
+    public boolean visit(MySqlSelectQueryBlock query) {
+        SQLExprTableSource from = (SQLExprTableSource) query.getFrom();
+        String tableName = from.getExpr().toString().replaceAll(" ", "");
+
+        if (from.getAlias() != null) {
+            tableScope.push(new Table(tableName, from.getAlias()));
+        } else {
+            from.setAlias(createAlias(tableName));
+            tableScope.push(new Table(tableName, from.getAlias()));
```

Comment:
duplicated line on #66 and #69, so it can be just
```java
if (from.getAlias() == null) {
    from.setAlias(createAlias(tableName));
}
tableScope.push(new Table(tableName, from.getAlias()));
return true;
```